cli/format.go

```diff
@@ -224,6 +245,31 @@ func walkFilesystem(ctx context.Context) func() error {
 		eg, ctx := errgroup.WithContext(ctx)
 		pathsCh := make(chan string, BatchSize)
 
+		// By default, we use the cli arg, but if the stdin flag has been set we force a filesystem walk
+		// since we will only be processing one file from a temp directory
+		walkerType := Cli.Walk
+
+		if Cli.Stdin {
+			walkerType = walk.Filesystem
+
+			// check we have only received one path arg which we use for the file extension / matching to formatters
+			if len(Cli.Paths) != 1 {
+				return fmt.Errorf("only one path should be specified when using the --stdin flag")
+			}
+
+			// read stdin into a temporary file with the same file extension
+			pattern := fmt.Sprintf("*%s", filepath.Ext(Cli.Paths[0]))
+			if file, err := os.CreateTemp("", pattern); err != nil {
+				return fmt.Errorf("failed to create a temporary file for processing stdin: %w", err)
+			} else if _, err = io.Copy(file, os.Stdin); err != nil {
+				return fmt.Errorf("failed to copy stdin into a temporary file")
+			} else {
+				Cli.Paths[0] = file.Name()
+			}
```

Comment:
This part is a bit weird because the if/else is not really that. It's mostly a series of statement that want to check for the err.
```suggestion
			if file, err := os.CreateTemp("", pattern); err != nil {
				return fmt.Errorf("failed to create a temporary file for processing stdin: %w", err)
			}
			defer os.Remove(f.Name()) // clean up
			if _, err = io.Copy(file, os.Stdin); err != nil {
				return fmt.Errorf("failed to copy stdin into a temporary file")
			}
			file.Close() // to flush to disk
			Cli.Paths[0] = file.Name()
```